### ISSUE TITLE
Error with Postgres 10 native partitioning and custom time intervals

### ISSUE THREAD

**Issue by andreaswachowski** [keithf4/pg_partman#200]
(These are my first attempts at using pg_partman, so I hope I have not overlooked something obvious.)

Using pg_partman 3.1.0 (alternatively the master branch) I experience the following error:

```
postgres_1  | 2017-11-19 12:14:24.929 UTC [29] LOG:  pg_partman master background worker master process initialized with role postgres
postgres_1  | 2017-11-19 12:14:24.991 UTC [31] LOG:  pg_partman dynamic background worker (dbname=db_devel...) dynamic background worker initialized with role postgres on database db_development
postgres_1  | 2017-11-19 12:14:25.167 UTC [31] ERROR:  case not found
postgres_1  |   CONTEXT: PL/pgSQL function show_partition_name(text,text) line 56 at CASE
postgres_1  |   SQL statement "SELECT suffix_timestamp                                    FROM partman.show_partition_name(v_row.parent_table, v_current_partition_timestamp::text)"
postgres_1  |   PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 177 at SQL statement
postgres_1  |   SQL statement "SELECT "partman".run_maintenance(p_analyze := true, p_jobmon := true)"
postgres_1  |   DETAIL:
postgres_1  |   HINT: CASE statement is missing ELSE part.
postgres_1  | 2017-11-19 12:14:25.167 UTC [31] CONTEXT:  PL/pgSQL function partman.run_maintenance(text,boolean,boolean,boolean) line 358 at RAISE
postgres_1  |   SQL statement "SELECT "partman".run_maintenance(p_analyze := true, p_jobmon := true)"
postgres_1  | 2017-11-19 12:14:25.174 UTC [1] LOG:  worker process: pg_partman dynamic background worker (dbname=db_devel...) (PID 31) exited with exit code 1
```

That happens with Postgres 10 (including `pg_jobmon`) and a custom interval of 2 minutes for a time-partitioned table:
```
SELECT partman.create_parent( 'public.db_development', 'timestamp', 'native', '2 minutes', NULL, 4, 'on', NULL, true, 'none', '', true, 'public.db_development_template');
```

`create_parent` exits without error. But when `run_maintenance` is called, the big `IF`-statement in `show_partition_name` seems to select the wrong `IF` branch (the one with the `CASE`-statement). I tried to hack around that and replaced

```
IF ( (v_control_type = 'time') OR (v_control_type = 'id' AND v_epoch <> 'none') )
     AND v_type <> 'time-custom'
```
with
```
IF ( (v_control_type = 'time') OR (v_control_type = 'id' AND v_epoch <> 'none') )
     AND v_type <> 'time-custom' AND (v_type <>'native' OR (v_partition_interval::interval IN('15 mins', '30 mins', '1 hour', '1 day', '1 week', '1 month', '3 months', '1 year')))
```

(and an analogous change for `ELSIF v_type = 'time-custom'` to enter that branch).

The problem now is that `show_partition_name` depends on values from `custom_time_partitions` when executing the following:
```
    IF partition_table IS NULL THEN
        SELECT max(upper(partition_range)) INTO v_max_range FROM @extschema@.custom_time_partitions WHERE parent_table = p_parent_table;
        SELECT min(lower(partition_range)) INTO v_min_range FROM @extschema@.custom_time_partitions WHERE parent_table = p_parent_table;
        IF p_value::timestamptz >= v_max_range THEN
```

But my `custom_time_partitions` table is empty. I gather that `create_partition_time` is responsible for initializing the values therein, but being new to all this, it's hard to analyze further, so I wanted to at least report this. Please let me know when any questions remain. Feedback is appreciated, thanks in advance!

**Comment by keithf4** [comment]
Thanks for reporting the issue. Will try and look into it soon.

**Comment by keithf4** [comment]
So native partitioning doesn't use the `custom_time_partitions` table, so it being empty is normal in this case. Problem is I didn't actually account for native partitioning having anything other than the default time partition intervals in `show_partition_name`. So this is going to take a bit for me to figure out a fix.

Thanks again for reporting the issue.

**Comment by andreaswachowski** [comment]
As far as I am concerned, there's no rush, I stumbled upon this in a private project. Thank you for your efforts and pg_partman!

**Comment by bbcnova** [comment]
I am just confirming that I ran into the same issue running run_maintenance() on a native partition set configured to create new partitions every 6 hours.

**Comment by keithf4** [comment]
This issue has been fixed in v3.1.1. Please let me know if it works ok for you now. Apologies again on the delay for this fix.
